### PR TITLE
CLOB `DeliverTx` audit cleanup

### PR DESCRIPTION
**Pull request by lucas-dydx**
Miscellaneous code cleanup from the `DeliverTx` flow audit.

Note this PR shouldn't contain any logic changes.

### CODE REVIEW

File: protocol/x/clob/keeper/process_operations_stateful_validation.go [hunk 1]
```diff
@@ -80,49 +80,6 @@ func (k Keeper) MustFetchOrderFromOrderId(
...
-
-	// Maker order cannot be FOK or IOC.
-	if makerOrder.GetTimeInForce() == types.Order_TIME_IN_FORCE_FILL_OR_KILL ||
-		makerOrder.GetTimeInForce() == types.Order_TIME_IN_FORCE_IOC {
-		return makerOrder, errorsmod.Wrapf(
-			types.ErrInvalidMatchOrder,
-			"Maker order %+v cannot be FOK or IOC.",
-			makerOrder.GetOrderTextString(),
-		)
-	}
```
Comment (lucas-dydx) [inline protocol/x/clob/keeper/process_operations_stateful_validation.go:122]: This validation is duplicated so I deleted it. Figured it's best to share it with the `CheckTx` flow by including it in the `MatchWithOrders.Validate` logic.

File: protocol/x/clob/keeper/process_operations_test.go [hunk 2]
```diff
@@ -953,6 +953,79 @@ func TestProcessProposerOperations(t *testing.T) {
 				},
 			},
 		},
+		"Fails when attempting to match order with invalid order side": {
```
Comment (lucas-dydx) [inline protocol/x/clob/keeper/process_operations_test.go:956]: Added a test to verify this fails.

File: protocol/x/clob/types/match_with_orders.go [hunk 3]
```diff
@@ -73,8 +73,9 @@ func (match MatchWithOrders) Validate() error {
 	}
 
 	// Make sure the maker order is not an IOC order.
-	if makerOrder.MustGetOrder().TimeInForce == Order_TIME_IN_FORCE_IOC {
-		return errors.New("IOC order cannot be matched as a maker order")
+	unwrappedMakerOrder := makerOrder.MustGetOrder()
+	if unwrappedMakerOrder.RequiresImmediateExecution() {
+		return errors.New("IOC / FOK order cannot be matched as a maker order")
```
Comment (lucas-dydx) [inline protocol/x/clob/types/match_with_orders.go:78]: I can update these errors to include the invalid order if anyone would prefer that.

File: protocol/x/clob/keeper/process_operations.go [hunk 4]
```diff
@@ -545,11 +545,8 @@ func (k Keeper) PersistMatchLiquidationToState(
 			FillAmount: satypes.BaseQuantums(fill.FillAmount),
 		}
 
-		if err := matchWithOrders.Validate(); err != nil {
-			return err
-		}
```
Comment (lucas-dydx) [inline protocol/x/clob/keeper/process_operations.go:550]: This is performed below in `ProcessSingleMatch` and shouldn't be duplicated.

**Review by jayy04 — APPROVED** [review]

File: protocol/x/clob/types/match_with_orders.go (hunk 3 above)
Comment (jakob-dydx) [reply protocol/x/clob/types/match_with_orders.go:78]: Including the order in the error sounds good to me

**Review by jakob-dydx — APPROVED** [review]

**Pull request merged**